### PR TITLE
feat(skip-link): add `<rh-skip-link>`

### LINKED ISSUE #1227: [feat] Create `<rh-skip-link>`
Create a Skip Nav element following [best practices](https://webaim.org/techniques/skipnav/).

### PR DESCRIPTION
**Pull request by adamjohnson**
## What I did

1. Added `<rh-skip-link>`

Closes #1227 

### TODO:

- [x] Replace `screenshot.png`.
- [x] Refine the docs / JSDoc.
- [x] Refine `README.md`.
- [x] Write tests.
- [x] Add changeset.
- [ ] Add Style, Guidelines, Overview, and Accessibility docs pages.

## Testing Instructions

1. View deploy preview demo.

### CODE REVIEW

File: elements/rh-skip-link/rh-skip-link.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
+:host {
+  display: block;
+  position: relative;
```
Comment (bennypowers) [inline elements/rh-skip-link/rh-skip-link.css:3]: This is fine but maybe we can put it on an inner container

**Review by bennypowers — CHANGES_REQUESTED** [review]
Looks good. A few tweaks and we're g2g

File: elements/rh-skip-link/rh-skip-link.css (hunk 1 above)
Comment (adamjohnson) [reply elements/rh-skip-link/rh-skip-link.css]: Moved these to `#rhds-skip-link-container`.

File: elements/rh-skip-link/rh-skip-link.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+::slotted(a) {
```
[the remark this answers is not in the document]
Comment (adamjohnson) [reply elements/rh-skip-link/rh-skip-link.css:6]: I added`!important` to every CSS property. 

Is this to enforce the design of the skip link?

File: .changeset/slow-drinks-grow.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@rhds/elements": minor
+---
+
+Added the `<rh-skip-link>` element.
```
[the remark this answers is not in the document]
Comment (adamjohnson) [reply .changeset/slow-drinks-grow.md]: Double check that I did this correctly. TIA!

File: elements/rh-skip-link/rh-skip-link.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-skip-link/rh-skip-link.css:6]: the slotted pseudo has no priority

File: .changeset/slow-drinks-grow.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (bennypowers) [reply .changeset/slow-drinks-grow.md]: See https://github.com/RedHat-UX/red-hat-design-system/blob/main/CHANGELOG.md#minor-changes


Tldr: emoji, phrasing, example

File: elements/rh-skip-link/rh-skip-link-lightdom-shim.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
+rh-skip-link:not(:defined) {
+  display: block !important;
+  position: relative !important;
+}
+
+rh-skip-link:not(:defined) > a {
+  background-color: var(--rh-color-surface-lightest, #ffffff) !important;
```
Comment (bennypowers) [inline elements/rh-skip-link/rh-skip-link-lightdom-shim.css:7]: important not needed in shim, only in shadow styles
other than that Liminal Gatorade Transfuses Memory

File: elements/rh-skip-link/rh-skip-link.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  border-inline-end:
+    var(--rh-border-width-sm, 1px)
+    solid
+    var(--rh-color-border-interactive-on-light, #0066cc) !important;
+  border-end-start-radius: var(--rh-border-radius-default, 3px) !important;
+  border-end-end-radius: var(--rh-border-radius-default, 3px) !important;
+  clip: rect(0, 0, 0, 0) !important;
+  color: var(--rh-color-interactive-blue-darker, #0066cc) !important;
+  font-family: var(--rh-font-family-heading, RedHatDisplay, "Red Hat Display", "Noto Sans Arabic", "Noto Sans Hebrew", "Noto Sans JP", "Noto Sans KR", "Noto Sans Malayalam", "Noto Sans SC", "Noto Sans TC", "Noto Sans Thai", Helvetica, Arial, sans-serif) !important;
+  font-weight: var(--rh-font-weight-heading-bold, 700) !important;
```
Comment (marionnegp) [inline elements/rh-skip-link/rh-skip-link.css:26]: Can `font-size: var(--rh-font-size-body-text-md, 1rem);` be added to make sure page styles don't change that too?

**Review by marionnegp — CHANGES_REQUESTED** [review]
Looks good! I left one comment.

File: elements/rh-skip-link/rh-skip-link.css (hunk 5 above)
Comment (adamjohnson) [reply elements/rh-skip-link/rh-skip-link.css:27]: Fixed in https://github.com/RedHat-UX/red-hat-design-system/pull/1515/commits/26ce8e3c5a16da825537fad19292c0e0f21346ed. 

Not sure why GitHub still marks this as "1 change requested"? 

@marionnegp perhaps you can mark this as resolved or hit "Dismiss Review". TIA.

**Review by marionnegp — APPROVED** [review]

**Review by zeroedin — APPROVED** [review]
Let's Go!  Touchdown Mountaineers!
